scripts/db-clone/clone.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+set -x
+
+
+export PGHOST=pathoplexus3.cluster-che6ccwa8oie.eu-central-1.rds.amazonaws.com
+export PGUSER=postgres
+
+print_usage() {
+    echo "Usage: $0 <action> <database> <file> [new_owner_username]"
+    echo "Actions:"
+    echo "  dump - Dump the specified database to a file"
+    echo "  load - Load a dump file into the specified database"
+    echo "Examples:"
+    echo "  $0 dump production temp_dump.sql"
+    echo "  $0 load staging temp_dump.sql staging_keycloak_user"
+}
+
+check_command() {
+    if [ $? -ne 0 ]; then
+        echo "Error: $1"
+        exit 1
+    fi
+}
+
+dump_database() {
+    local db_name="$1"
+    local dump_file="$2"
+    echo "Dumping $db_name database to $dump_file..."
+    pg_dump "$db_name" > "$dump_file"
+    check_command "Failed to dump $db_name database"
+    echo "Database dump completed successfully!"
+    echo "Dump file: $dump_file"
+}
+
+prepare_database() {
+    local db_name="$1"
+    echo "Preparing database $db_name..."
+    psql -d postgres <<EOF
+        SELECT pg_terminate_backend(pid) 
+        FROM pg_stat_activity 
+        WHERE datname = '$db_name' AND pid <> pg_backend_pid();
+        REVOKE CONNECT ON DATABASE $db_name FROM public;
+        ALTER DATABASE $db_name CONNECTION LIMIT 0;
+EOF
+    check_command "Failed to prepare database $db_name"
+}
+
+restore_limited_access() {
+    local db_name="$1"
+    echo "Restoring limited access for postgres user only on $db_name..."
+    psql -d postgres <<EOF
+        SELECT pg_terminate_backend(pid) 
+        FROM pg_stat_activity 
+        WHERE datname = '$db_name' AND pid <> pg_backend_pid();
+        REVOKE CONNECT ON DATABASE $db_name FROM public;
+        ALTER DATABASE $db_name CONNECTION LIMIT 1;
+EOF
+    check_command "Failed to restore limited access to $db_name"
+}
+
+restore_full_access() {
+    local db_name="$1"
+    echo "Restoring full access for all users on $db_name..."
+    psql -d postgres <<EOF
+        GRANT CONNECT ON DATABASE $db_name TO public;
+        ALTER DATABASE $db_name CONNECTION LIMIT -1;
+EOF
+    check_command "Failed to restore full access to $db_name"
+}
+
+change_ownership() {
+    local db_name="$1"
+    local new_owner="$2"
+    echo "Changing ownership to $new_owner"
+    psql -d "$db_name" <<EOF
+    DO \$\$
+    DECLARE
+        r RECORD;
+    BEGIN
+        -- Change ownership of tables
+        FOR r IN (SELECT schemaname, tablename FROM pg_tables WHERE schemaname = 'public')
```

Comment:
Couple of things to think about:

1. If in the future we add anything else like Materialized Views we'll need to remember to add them here. In an ideal world we might use another approach that didn't require us to iterate over the objects if that was possible. E.g. if they were created by their eventual owner? But this may be complicated
2. I think @anna-parker already uses a second schema in addition to `public` ?

Reply:
yes - currently ena-submission is disable so that means the schema has not been created yet on prod/staging but as soon as it has been we will have an additional schema called `ena-submission`